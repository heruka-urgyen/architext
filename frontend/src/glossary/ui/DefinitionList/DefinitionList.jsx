import {DefinitionBlock} from "../DefinitionBlock"
import {Title} from "../Title"

function DefinitionList({entries, handleTitleClick}) {
  return (
    entries.map(terms => {
      if (terms.length === 0) {
        return null
      }
      const [[term]] = terms
      const definitions = terms.map(([_, def]) => def)

      const titleNode = (
        <Title term={term} onClick={handleTitleClick} />
      )

      return (
        <DefinitionBlock
          key={term}
          titleNode={titleNode}
          definitions={definitions}
        />
      )
    })
  )
}

export {DefinitionList}
