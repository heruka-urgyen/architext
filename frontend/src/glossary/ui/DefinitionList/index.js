export {DefinitionList} from "./DefinitionList.jsx"
