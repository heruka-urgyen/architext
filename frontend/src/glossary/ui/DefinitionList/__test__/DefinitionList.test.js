/**
 * @jest-environment jsdom
 */

import "regenerator-runtime/runtime"
import {render} from "@testing-library/react"
import "@testing-library/jest-dom"

import {DefinitionList} from "../DefinitionList"

describe("DefinitionList", () => {
  test("matches snapshot", () => {
    const entries = [
      [["term 1", "def 1-1", "def 1-2"]],
      [["term 2", "def 2-1", "def 2-2"]],
    ]

    const handleTitleClick = jest.fn(_ => _)

    const screen = render(
      <DefinitionList entries={entries} handleTitleClick={handleTitleClick} />,
    )

    expect(screen).toMatchSnapshot()
  })
})
