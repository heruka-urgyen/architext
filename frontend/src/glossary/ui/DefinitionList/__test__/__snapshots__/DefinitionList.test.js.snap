// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefinitionList matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3>
        <span
          class="term searchable"
        >
          term
        </span>
        <span
          class="term searchable"
        >
          1
        </span>
      </h3>
      <ul>
        <li>
          def 1-1
        </li>
      </ul>
      <h3>
        <span
          class="term searchable"
        >
          term
        </span>
        <span
          class="term searchable"
        >
          2
        </span>
      </h3>
      <ul>
        <li>
          def 2-1
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h3>
      <span
        class="term searchable"
      >
        term
      </span>
      <span
        class="term searchable"
      >
        1
      </span>
    </h3>
    <ul>
      <li>
        def 1-1
      </li>
    </ul>
    <h3>
      <span
        class="term searchable"
      >
        term
      </span>
      <span
        class="term searchable"
      >
        2
      </span>
    </h3>
    <ul>
      <li>
        def 2-1
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
